game-backend/src/helpers/gameObjects.ts

```diff
@@ -11,3 +13,33 @@ export const countUnits = (state: State): number =>
 
 export const countFood = (state: State): number =>
   countGameObjectsPerType(state, GAME_OBJECT_TYPES.FOOD);
+
+export const generateFreePosition = (state: State): Position => {
+  const fields: Position[] = new Array(ARENA_WIDTH)
+    .fill(undefined)
+    .reduce(
+      (acc, _, i) => [
+        ...acc,
+        ...new Array(ARENA_HEIGHT)
+          .fill(undefined)
+          .map((omit, j) => ({ x: i, y: j })),
+      ],
+      []
+    );
+  const occupiedFields: Position[] = state.objectIds.map(
+    id => state.objectMap[id].position
+  );
+  const freeFields = fields.filter(
+    field =>
+      !!occupiedFields.find(
```

Comment:
Watch out, this has bad runtime performance due to nested loop.
Why not just create a `map` of occupiedFields and access them by key (key e.g. `'${x}:${y}'`)?